shared/src/main/scala/crystal/viewF.scala

```diff
@@ -31,15 +30,15 @@ final class ViewF[F[_]: Async, A](val get: A, val mod: (A => A) => F[Unit])
 
   // In a ViewF, we can derive modAndGet. In ViewOptF and ViewListF we have to pass it, since their
   // mod functions have no idea of the enclosing structure.
-  val modAndGet: (A => A) => F[A] = f => {
-    Sync[F].delay(Promise[A]()).flatMap { p =>
+  val modAndGet: (A => A) => F[A] = f =>
+    Async[F].async { cb =>
```

Comment:
Cool, I tried to do something like this but with `Defered` and couldn't make it work
e